internal/services/managedredis/managed_redis_resource.go

```diff
@@ -526,6 +556,9 @@ func (r ManagedRedisResource) Update() sdk.ResourceFunc {
 						if metadata.ResourceData.HasChange("default_database.0.eviction_policy") {
 							dbParams.Properties.EvictionPolicy = pointer.ToEnum[databases.EvictionPolicy](state.DefaultDatabase[0].EvictionPolicy)
 						}
+						if metadata.ResourceData.HasChange("default_database.0.persistence") {
+							dbParams.Properties.Persistence = expandPersistence(state.DefaultDatabase[0].Persistence)
+						}
```

Comment:
Could it happen if the updated config removes the `default_database.0.persistence`? In this case, dereferencing the 0th index of DefaultDatabase will panic.

Reply:
Yes default_database can be removed, but it is already handled above. See `case dbLen(old) == 1 && dbLen(new) == 0:` at line 508.